contrib/extensions/xss/src/main/java/org/apache/sling/xss/impl/XSSAPIImpl.java

```diff
@@ -46,6 +58,12 @@
 
     private static final Pattern PATTERN_AUTO_DIMENSION = Pattern.compile("['\"]?auto['\"]?");
 
+    private static SAXParserFactory factory = SAXParserFactory.newInstance();
```

Comment:
Wouldn't this be better implemented as a field in activate which is then set to null on deactivate? This minimises the potential for pinning classes in memory once the bundle is updated